sql/core/src/test/scala/org/apache/spark/sql/DataFrameSuite.scala

```diff
@@ -2943,6 +2943,34 @@ class DataFrameSuite extends QueryTest
       .withSequenceColumn("default_index").collect().map(_.getLong(0))
     assert(ids.toSet === Range(0, 10).toSet)
   }
+
+  test("SPARK-35320 DataFrame read in Json format should fail if the schema provided " +
```

Comment:
Could we make the test title simpler? e.g.) Reading JSON with string key in a map should fail